fairscale/optim/oss.py

```diff
@@ -546,6 +547,12 @@ def _sync_param_groups(source: List[Dict[Any, Any]], destination: List[Dict[Any,
     def _broadcast_params(self) -> None:
         """Helper function to broadcast all the parameters from a given device"""
 
+        # if NCCL broadcasts will be done in an independent stream
+        # make sure that prior compute work is complete
+        if self.backend == dist.Backend.NCCL:
+            for device in self.per_device_params.keys():
+                torch.cuda.synchronize(device=device)
```

Comment:
this syncs the specified GPU and CPU, right? Does this actually sync streams? Is there a way to have a unit test and trigger this and ensure the fix is right?

Reply:
it should just sync all the streams, which does not imply necessarily a cpu/gpu sync I think. It's a good question, I was also wondering about a unit test, I'll try to make up a fake optimizer which waits on GPU for a long time

Reply:
so it seems that it's somehow not needed for NCCL actually, because pytorch is doing the same (but it's not doing this for gloo..)